src/trusted/service_runtime/sel_ldr_standard.c

```diff
@@ -735,8 +798,8 @@ int NaClCreateMainThread(struct NaClApp     *nap,
     }
   }
   envv_len = 0;
-  argv_len = malloc(argc * sizeof argv_len[0]);
-  envv_len = malloc(envc * sizeof envv_len[0]);
+  argv_len = !argc ? NULL : malloc(argc * sizeof argv_len[1]);
+  envv_len = !envc ? NULL : malloc(envc * sizeof envv_len[0]);
```

Comment:
@SantiagoTorres  It's kinda confusing but there is a point: basically it is to avoid `free(<uninitialized pointer>)` in the cleanup path. Hopefully this fix makes it clearer: 4ccc5bc

Reply:
yeah, I think 4ccc5bc pretty much fixes it. Thanks!